### PR TITLE
Hipp 869 provide supporting information page

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: test/controllers/application/ProvideSupportingInformationControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    val controller = playApplication.injector.instanceOf[ProvideSupportingInformationController]
+    Fixture(playApplication, apiHubService, accessRequestSessionRepository, controller)
+  }
+
+  private def buildUserAnswers(application: Application): UserAnswers = {
+    UserAnswers(id = FakeUser.userId, lastUpdated = clock.instant())
+      .set(AccessRequestApplicationIdPage, application).toOption.value
+
+//    userAnswers.set(RequestProductionAccessPage, Set(Accept)).toOption.value
```
Comment (PaulCDurham) [inline test/controllers/application/ProvideSupportingInformationControllerSpec.scala:145]: Remove this?

File: test/controllers/application/ProvideSupportingInformationControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            val captor: ArgumentCaptor[UserAnswers] = ArgumentCaptor.forClass(classOf[UserAnswers])
+            verify(fixture.accessRequestSessionRepository).set(captor.capture())
+            val userAnswers: UserAnswers = captor.getValue
+            userAnswers.get(ProvideSupportingInformationPage) mustEqual Some("blah")
+            // TODO in HIP 870:
+            // redirectLocation(result) mustBe Some(controllers.application.routes.Whatever.onPageLoad().url)
+          }
+      }
+    }
+  }
```
Comment (PaulCDurham) [inline test/controllers/application/ProvideSupportingInformationControllerSpec.scala:94]: Add a test to show that the page display correctly when the question has already been answered (ie filling the form from user answers works)

File: test/controllers/application/ProvideSupportingInformationControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
```
Comment (PaulCDurham) [inline test/controllers/application/ProvideSupportingInformationControllerSpec.scala:95]: Add a test to show that an invalid submission has a 400 Bad Request response (ie filing the form using formWithErrors)

**Review by PaulCDurham — CHANGES_REQUESTED** [review]
Couple of missing tests and some trivial stuff

File: app/controllers/application/RequestProductionAccessController.scala [hunk 4]
```diff
@@ -17,44 +17,73 @@
...
-      applicationApiBuilder.build(request.application).map {
-        case Right(applicationApis) => form.bindFromRequest().fold(
-          formWithErrors =>
-            Ok(view(formWithErrors, mode, request.application, applicationApis, Some(request.identifierRequest.user))),
-          value => Redirect(controllers.routes.JourneyRecoveryController.onPageLoad()))
-        case Left(result) => result
-      }
+      form.bindFromRequest().fold(
+          formWithErrors => {
+            Console.println(s"formWithErrors: $formWithErrors")
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply app/controllers/application/RequestProductionAccessController.scala:78]: Ok

File: test/controllers/application/ProvideSupportingInformationControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import play.api.{Application => PlayApplication}
+import repositories.AccessRequestSessionRepository
+import services.ApiHubService
+import utils.{HtmlValidation, TestHelpers}
+import views.html.application.ProvideSupportingInformationView
+
+import java.time.{Clock, Instant, ZoneId}
+import scala.concurrent.Future
+
+class ProvideSupportingInformationControllerSpec extends SpecBase with MockitoSugar with TestHelpers with HtmlValidation {
```
[the remark this answers is not in the document]
Comment (llmikeyj) [reply test/controllers/application/ProvideSupportingInformationControllerSpec.scala:42]: Ok

File: test/controllers/application/ProvideSupportingInformationControllerSpec.scala (hunk 1 above)
Comment (llmikeyj) [reply test/controllers/application/ProvideSupportingInformationControllerSpec.scala:145]: Ok

File: test/controllers/application/ProvideSupportingInformationControllerSpec.scala (hunk 2 above)
Comment (llmikeyj) [reply test/controllers/application/ProvideSupportingInformationControllerSpec.scala:94]: Ok

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**